packages/playwright/src/runner/projectUtils.ts

```diff
@@ -24,26 +24,31 @@ import { createFileMatcher } from '../util';
 const readFileAsync = promisify(fs.readFile);
 const readDirAsync = promisify(fs.readdir);
 
+// The difference to forceRegExp is that we want to match the whole string.
+function forceBoundedRegExp(pattern: string): RegExp {
+  const match = pattern.match(/^\/(.*)\/([gi]*)$/);
+  if (match)
+    return new RegExp(match[1], match[2]);
+  return new RegExp(`^${pattern}$`, 'gi');
```

Comment:
Why do we force full string match, rather than do a substring one?

Reply:
In order to not break backwards compatibility.